src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/Razor/ExtractToNewComponentCodeActionProviderTest.cs

```diff
@@ -98,4 +248,21 @@ private static RazorCodeActionContext CreateRazorCodeActionContext(VSCodeActionP
 
         return context;
     }
+
+
+    private static void AddMultiPointSelectionToContext(ref RazorCodeActionContext context, TextSpan selectionSpan)
+    {
+        var sourceText = context.CodeDocument.GetSourceText();
+        var startLinePosition = sourceText.Lines.GetLinePosition(selectionSpan.Start);
+        var startPosition = new Position(startLinePosition.Line, startLinePosition.Character);
+
+        var endLinePosition = sourceText.Lines.GetLinePosition(selectionSpan.End);
+        var endPosition = new Position(endLinePosition.Line, endLinePosition.Character);
+
+        context.Request.Range = new Range
+        {
+            Start = startPosition,
+            End = endPosition
+        };
```

Comment:
Note: You don't have the new extension methods or `VsLspFactory` that merged to main with #10670 (and will be updated eventually with #10682), but you should be able to replace this code with a single helper once you have the latest main.